controllers/metal3.io/baremetalhost_controller_test.go

```diff
@@ -1059,6 +1098,52 @@ func TestProvision(t *testing.T) {
 	)
 }
 
+// TestProvisionCustomDeploy ensures that the Provisioning.CustomDeploy portion
+// of the status block is filled in for provisioned hosts.
+func TestProvisionCustomDeploy(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Spec.CustomDeploy = &metal3v1alpha1.CustomDeploy{
+		Method: "install_everything",
+	}
+	host.Spec.Online = true
+	r := newTestReconciler(host)
+
+	tryReconcile(t, r, host,
+		func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
+			t.Logf("custom deploy: %v", host.Spec.CustomDeploy)
+			t.Logf("provisioning custom deploy: %v", host.Status.Provisioning.CustomDeploy)
+			t.Logf("provisioning state: %v", host.Status.Provisioning.State)
+			return host.Status.Provisioning.CustomDeploy != nil && host.Status.Provisioning.CustomDeploy.Method != ""
```

Comment:
Just to be sure: `host.Status.Provisioning.CustomDeploy.Method == "install_everything"`

Also, to be on the safe side, it could be worth adding a check on the state, ie `.. && host.Status.Provisioning.State == metal3v1alpha1.StateProvisioned`